internal/client/client.go

```diff
@@ -81,10 +81,10 @@ func getTokenFromCreds(services *disco.Disco, hostname svchost.Hostname) string
 //
 // Internally, this function caches configured clients using the specified
 // parameters
-func GetClient(tfeHost, token string, insecure bool) (*tfe.Client, error) {
-	config, err := configure(tfeHost, token, insecure)
+func GetClient(tfeHost, token string, insecure bool) (*tfe.Client, bool, error) {
+	config, sendCredentialDeprecationWarning, err := configure(tfeHost, token, insecure)
```

Comment:
Instead of drilling a third return value through to `configure()`, I'd probably add a new field to `ClientConfiguration`... something like, `CredentialSource`... then combine that outcome with `providerRunningInCloud` in the provider code, closer to the diagnostic

Reply:
I think that would make it much more flexible/readable as well as moving the different concerns together